main.py

```diff
@@ -38,52 +40,60 @@ def __init__(self):
             exit('Check if the configuration is set right')
 
     def _run(self):
-        for submission in self.reddit.subreddit(self.config.SUBREDDIT).stream.submissions():
-            print(submission.title + " - http://reddit.com" + submission.permalink)
-
+        for submission in self._get_submissions():
             exceeds_threshold = False
+            parsed_submission = TitleParser.create_parsed_submission_from_submission(submission)
 
-            if self.database.known_submission(submission) is True:
-                continue
-
-            if submission.is_self is True:
-                self.database.save_submission(submission)
-                # We currently don't do anything further with self posts.  Move to the next post.
-                continue
-
-            try:
-                parsed_submission = TitleParser.create_parsed_submission_from_submission(submission)
-            except LookupError:
-                # Can't find artist from submission name, skipping
-                self.database.log_error_causing_submission(None, submission, traceback.format_exc(), True)
+            if ExcludedArtists.is_excluded(parsed_submission):
```

Comment:
This means we won't save that submission to the DB.  Doubt it'll matter much as we would only see it again on restart and we know we will fail to parse it and continue anyway.

Reply:
Solved this by adding saving to `_get_submissions`